### PR TITLE
Disallow building source container from source container image build,

### PR DESCRIPTION
**Pull request by rcerven**
and also disallow any other than 'image' build type

* OSBS-8319

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Pull request includes link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 1]
```diff
@@ -1029,11 +1029,22 @@ def get_source_build_info(self, build_id, build_nvr):
                 )
             raise koji.BuildError(err_msg)
 
+        build_type = self.session.getBuildType(koji_build['build_id'], strict=True)
+        if 'image' not in build_type:
+            type_string = list(build_type.keys())[0]
```
Comment (athos-ribeiro) [inline koji_containerbuild/plugins/builder_containerbuild.py]: should we list all the types instead?

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 2]
```diff
@@ -1029,11 +1029,22 @@ def get_source_build_info(self, build_id, build_nvr):
                 )
             raise koji.BuildError(err_msg)
 
+        build_type = self.session.getBuildType(koji_build['build_id'], strict=True)
```
Comment (athos-ribeiro) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Can't we infer the type from `koji_build`?

File: tests/test_builder_containerbuild.py [hunk 3]
```diff
@@ -1206,40 +1209,65 @@ def test_build_nvr_exists(self, tmpdir, orchestrator, df_release, param_release,
...
         different_build_id = 54321
 
+        provided_nvr = create_args.get('koji_build_nvr')
+        provided_id = create_args.get('koji_build_id')
+        provided_name = None
+        if provided_nvr:
+            provided_name = provided_nvr.rsplit('-', 2)[0]
+
         session = self._mock_session(last_event_id, koji_task_id)
+        (session).should_receive('getBuildType').and_return({build_type: {}})
```
Comment (athos-ribeiro) [inline tests/test_builder_containerbuild.py]: Could we add a test with multiple build types being returned (this is the case for operators)

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 4]
```diff
@@ -1029,11 +1029,22 @@ def get_source_build_info(self, build_id, build_nvr):
...
+        if 'image' not in build_type:
+            type_string = list(build_type.keys())[0]
+            err_msg = ('koji build {} is {} build, source container '
+                       'build needs image build'.format(koji_build['nvr'], type_string))
+            raise koji.BuildError(err_msg)
+        elif koji_build['name'].endswith('-source'):
```
Comment (twaugh) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Same comment as in https://github.com/containerbuildsystem/atomic-reactor/pull/1317#discussion_r352060822 -- I think we should look for a source_for_nvr key to decide whether it is a a source container image build.

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 1 above)
Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py]: sure

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 2 above)
Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py]: yeah

File: tests/test_builder_containerbuild.py (hunk 3 above)
Comment (rcerven) [reply tests/test_builder_containerbuild.py]: yes

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 4 above)
Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py]: yes will change

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 5]
```diff
@@ -1029,11 +1029,23 @@ def get_source_build_info(self, build_id, build_nvr):
...
+        elif 'sources_for_nvr' in koji_build['extra']['image']:
+            err_msg = ('koji build {} is source container build, source container can not '
+                       'use source container build image'.format(koji_build['nvr']))
+            raise koji.BuildError(err_msg)
+
         if not build_id:
             build_id = koji_build['build_id']
         if not build_nvr:
             build_nvr = koji_build['nvr']
-        component = "%s-source" % koji_build['package_name']
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Unrelated change

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py]: package_name & name have same value

but since we decided in atomic-reactor to use 'name' I wanted to make it consistent

Comment (MartinBasti) [reply koji_containerbuild/plugins/builder_containerbuild.py]: should be a separate commit with proper description then for better git archaeology

**Comment by rcerven** [comment]
rn not updated: fix for unreleased feature
